vllm/entrypoints/openai/serving_engine.py

```diff
@@ -994,6 +1001,9 @@ async def _normalize_prompt_tokens_to_input(
         else:
             input_ids = prompt_ids[-truncate_prompt_tokens:]
 
+        # Validate lengths before any decode to avoid extra work on very long inputs.
+        self._validate_length_only(request, len(input_ids))
```

Comment:
If the validation is being called here, it doesn't need to be called again below (at end of this method)